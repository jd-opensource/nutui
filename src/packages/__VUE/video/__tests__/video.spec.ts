import { mount } from '@vue/test-utils';
import Video from '../index.vue';

describe('Video', () => {
  test('base Video', () => {
    const wrapper = mount(Video);
    const rate = wrapper.find('.nut-video');
    expect(rate.exists()).toBe(true);
  });
  test('should be displayed after setting the source src and type', () => {
    const wrapper = mount(Video, {
      props: {
        source: {
          src: 'xxx.mp4',
          type: 'video/mp4'
        },
        options: {
          controls: true
        }
      }
    });
    expect(wrapper.html()).toContain('src');
    expect(wrapper.html()).toContain('type');
    expect(wrapper.html()).toContain('controls');
  });
  test('should be displayed after setting the options autoplay and muted and loop', () => {
    const wrapper = mount(Video, {
      props: {
        options: {
          autoplay: true,
          muted: true,
          loop: true
        }
      }
    });
    expect(wrapper.html()).toContain('autoplay');
    expect(wrapper.html()).toContain('muted');
    expect(wrapper.html()).toContain('loop');
  });
  test('should be displayed after setting the options poster and playsinline', () => {
    const wrapper = mount(Video, {
      props: {
        options: {
          poster: 'xxx.png',
          playsinline: true
        }
      }
    });
    expect(wrapper.html()).toContain('xxx.png');
    expect(wrapper.html()).toContain('playsinline');
  });
  test('should be displayed after setting the click', () => {
    const wrapper = mount(Video);
    const _html1 = wrapper.find('.show-control');
    expect(_html1.exists()).toBe(true);
    wrapper.find('.control-play-btn').trigger('click');
    const _html2 = wrapper.find('.hide-control');
    expect(_html2.exists()).toBe(false);
  });
});
