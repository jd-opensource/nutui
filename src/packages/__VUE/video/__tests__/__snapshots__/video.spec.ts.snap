// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Video: base info 1`] = `
"<div class="nut-video"><video class="nut-video-player" autoplay="" loop="" controls="" playsinline="true" webkit-playsinline="true" x5-video-player-type="h5-page" x5-video-player-fullscreen="false">
    <source src="https://storage.jd.com/about/big-final.mp4" type="video/mp4">
  </video>
  <!--v-if-->
  <!--v-if-->
  <div class="nut-video-controller nut-video-controller--show" style="display: none;">
    <div class="nut-video-controller__playbtn"></div>
    <div class="nut-video-controller__now">00:00</div>
    <div class="nut-video-controller__progress">
      <div class="nut-video-controller__progress-value">
        <div class="buffered" style="width: 0%;"></div>
        <div class="nut-video-controller__ball" style="transform: translate3d(0px, -50%, 0);">
          <div class="nut-video-controller__ball-move"></div>
        </div>
        <div class="nut-video-controller__played"></div>
      </div>
    </div>
    <div class="nut-video-controller__total">00:00</div>
    <div class="nut-video-controller__volume muted"></div>
    <div class="nut-video-controller__full"></div>
  </div><!-- 错误弹窗 -->
  <div class="nut-video-error" style="display: none;">
    <p class="nut-video-error-tip">视频加载失败</p>
    <p class="nut-video-error-retry">点击重试</p>
  </div>
</div>"
`;
